File: main/java/nullbots/nullmage/ChooseSpell.java
package nullbots.nullmage;

import com.runemate.game.api.script.framework.tree.BranchTask;
import com.runemate.game.api.script.framework.tree.TreeTask;

public class ChooseSpell extends BranchTask {

    private SelectAlch selectAlch = new SelectAlch();
    private SelectCurse selectCurse = new SelectCurse();

    @Override
    public TreeTask successTask() {
        return selectCurse;
    }

    @Override
    public TreeTask failureTask() {
        return selectAlch;
    }

    @Override
    public boolean validate() {
        return NullMage.lastSpell;
    }
}

File: kotlin/archbots/archmage/treebot/branches/ChooseSpell.kt
package archbots.archmage.treebot.branches

import archapi.treebot.framework.BranchTask
import archapi.treebot.framework.TreeTask
import archbots.archmage.treebot.leaves.SelectAlch
import archbots.archmage.treebot.leaves.SelectCurse

class ChooseSpell : BranchTask() {

    private val selectAlch = SelectAlch()
    private val selectCurse = SelectCurse()

    override fun successTask(): TreeTask {
        return selectCurse
    }

    override fun failureTask(): TreeTask {
        return selectAlch
    }

    override fun validate(): Boolean {
        return true;
    }
}
